0.preprocess-sites/4.image-and-segmentation-qc.py

```diff
@@ -504,14 +508,48 @@
         )
 
     image_corr_df = (
-        pd.concat(image_corr_list).drop_duplicates(subset="site_location").reset_index()
+        pd.concat(image_corr_list).drop_duplicates(subset="site").reset_index()
+    )
+
+    image_corr_df["min_corr_value"] = (
+        image_corr_df[corr_qc_cols].astype(float).max(skipna=True, axis=1)
     )
 
     for col in corr_qc_cols:
         image_corr_df.loc[(image_corr_df[col] >= correlation_threshold), col] = "pass"
 
+    for col in corr_qc_cols:
```

Comment:
I am confused about what this is doing. Also, is there a reason to loop over `corr_qc_cols` twice? Or can these operations be combined in the same loop?

Reply:
There is no need to loop twice. I've combined them now. 
I'm doing some perhaps un-intuitive manipulation so that the resulting .csv is easy for a human to extract information at a glance. The goal:
We have correlation scores for each site for each pairwise image combo. 
It looks for anything where the correlation is good enough (>= correlation_threshold) and changes the value to "pass" since if the sites are aligned, we don't really care at a glance how good the good is.
Then it looks for any column where there are no alignment problems (if all "pass" then len(image_corr_df[col].unique()) == 1) and drops those columns.
So the resulting spreadsheet only contains sites with alignment problems and only scores for image pairs that have problems.

Reply:
if all correlations fail, will `len(image_corr_df[col].unique()) == 1` be `True`?

Reply:
no. there is effectively no way that the correlations would all fail with the same score.